Azure/azure-cli
# [App Config] `az appconfig`: Add premium SKU support

Related command

az appconfig create/update

Description

Adds new option to --sku argument in az appconfig update/create

Adds new arguments to az appconfig create i.e --replica-name, --replica-location and --no-replica
Either of the new arguments will be required when creating a premium tier app configuration store.

A user should either specify --no-replica or create a replica using --replica-name and --replica-location.

Testing Guide

Create premium tier store without replica: ( `az appconfig create -n <storename> -l <yourlocation> --sku Premium --no-replica`)
Create premium tier store with replica: ( `az appconfig create -n <storename> -l <yourlocation> --sku Premium --replica-name <yourReplica> --replica-location <replicaLocation>`)
History Notes

This is the same change as this PR:
https://github.com/Azure/azure-cli/pull/29417

New updates not in the previous PR:
Added validation on the sku property for `az appconfig create`



---

This checklist is used to make sure that common guidelines for a pull request are followed.
